storage/storage-interface/src/lib.rs

```diff
@@ -16,6 +16,38 @@ use libra_types::{
 use serde::{Deserialize, Serialize};
 use std::sync::Arc;
 use storage_proto::StartupInfo;
+use thiserror::Error;
+
+#[derive(Debug, Deserialize, Error, PartialEq, Serialize)]
+pub enum Error {
+    #[error("Service error: {:?}", error)]
+    ServiceError { error: String },
+
+    #[error("Serialization error: {0}")]
+    SerializationError(String),
```

Comment:
why the difference between the two variants?

Reply:
The thing we really need is a serializable Error type since `anyhow::Error` is not serializable. The two variants are just for a little bit details.